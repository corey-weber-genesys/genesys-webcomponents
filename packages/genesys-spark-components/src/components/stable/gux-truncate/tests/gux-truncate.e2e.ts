import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { a11yCheck } from '../../../../test/e2eTestUtils';

async function newNonrandomE2EPage(
  {
    html
  }: {
    html: string;
  },
  lang: string = 'en'
): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(`<div lang=${lang}>${html}</div>`);
  await page.waitForChanges();
  await page.addScriptTag({
    path: '../../node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-truncate', () => {
  describe('#render', () => {
    [
      '<gux-truncate style="width: 40px">Some text to truncate</gux-truncate>',
      '<gux-truncate style="width: 40px"><span>Some text to truncate in a span</span></gux-truncate>',
      '<gux-truncate style="width: 40px"><div>Div <span>with a span</span> inside</div></gux-truncate>',
      '<gux-truncate style="width: 40px" max-lines="3">This is a long text that should be truncated after three lines of wrapped text</gux-truncate>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-truncate');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
