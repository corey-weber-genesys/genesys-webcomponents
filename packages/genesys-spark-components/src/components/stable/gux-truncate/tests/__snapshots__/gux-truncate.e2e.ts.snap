// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-truncate #render should render component as expected (1) 1`] = `"<gux-truncate hydrated=\\"\\">Some text to truncate</gux-truncate>"`;

exports[`gux-truncate #render should render component as expected (2) 1`] = `"<gux-truncate hydrated=\\"\\"><span>Some text to truncate in a span</span></gux-truncate>"`;

exports[`gux-truncate #render should render component as expected (3) 1`] = `"<gux-truncate hydrated=\\"\\"><div>Div <span>with a span</span> inside</div></gux-truncate>"`;

exports[`gux-truncate #render should render component as expected (4) 1`] = `"<gux-truncate max-lines=\\"3\\" hydrated=\\"\\">This is a long text that should be truncated after three lines of wrapped text</gux-truncate>"`;
